src/Elasticsearch/Metadata/ElasticsearchDocument.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Elasticsearch\Metadata;
+
+use ApiPlatform\Metadata\PersistenceMeansInterface;
+
+class ElasticsearchDocument implements PersistenceMeansInterface
```

Comment:
What about `ElasticsearchStateOptions` ? See comment below about naming I don't understand the interface name nor why this is a Document.

Reply:
The resource is persisted by the means of an elasticsearch document.